### ISSUE TITLE
autoPlay on android and iOs 

### ISSUE THREAD

**Issue by irongomme** [clappr/clappr#404]
Unfortunately, and dispites the old issues (#30 & #214), 
the autoPlay option doesn't trigger play on iOs (8.1.2 for me) and chrome on android (4.3 for me)

I've noticed this only on hls with html5video playback , i haven't tested others streaming types ...

**Comment by irongomme** [comment]
Ok, i've got one answer for iOs : 

from https://developer.apple.com/library/safari/documentation/AudioVideo/Conceptual/Using_HTML5_Audio_Video/Device-SpecificConsiderations/Device-SpecificConsiderations.html :

> In Safari on iOS (for all devices, including iPad), where the user may be on a cellular network and be charged per data unit, preload and autoplay are disabled. No data is loaded until the user initiates it. This means the JavaScript play() and load() methods are also inactive until the user initiates playback, unless the play() or load() method is triggered by user action. In other words, a user-initiated Play button works, but an onLoad="play()" event does not.

Now i continue my investigations for android ....

**Comment by irongomme** [comment]
It seems to be the same on Android, both for chrome and native browser.

Unless someone's got a trick for that , i believe there are no solutions to resolve this problem ! :hankey:

**Comment by towerz** [comment]
@irongomme As far as I can tell, most mobile browsers on iOS and Android rely on user interaction to trigger playback. In Safari on iOS, it's even worse, because it even ignores the preload attribute on the video tag.

A solution from an UX standpoint is to present a list of videos for the user to select, and have a hidden player instance. When the user taps an element on this list, you could call load() on the player and make it visible. This way the video will start automatically, but only after an user interaction.

**Comment by irongomme** [comment]
@towerz this is a good idea, but if i'm not wrong, Clappr doesn't allow to be instanciate without source !?

**Comment by towerz** [comment]
@irongomme if you instantiate with no source parameter or with an empty string as source, Clappr will instantiate the player using a NoOp playback. You can use the load method normally after that, like this:
```
var playerElement = document.getElementById("player-wrapper");
var player = new Clappr.Player({
  mute: true,
  height: 360,
  width: 640
});
player.attachTo(playerElement);
```
Then, on user tap:
```
player.load('http://cdn.clappr.io/highline.mp4');
player.play();
```

**Comment by towerz** [comment]
@irongomme I don't think we can do much for this in Clappr itself.

Marking as wontfix, but if anyone finds a solution for this we can reopen it.

**Comment by FDiskas** [comment]
This is done for security reasons. Video won't play on mobile devices. Only after user interaction it's possible to trigger the player play event.
But there is still another issue. Then the source of the player is changed the video stops playing. Needs to hit the play button again.
For example "videojs" - you can change the stream source and video still be auto played. Looks like it's possible to fix.

**Comment by kslimani** [comment]
@FDiskas this issue has been fixed in #1318
